src/OpenTelemetry/Metrics/AggregatorStore.cs

```diff
@@ -39,54 +42,178 @@ internal class AggregatorStore
         private DateTimeOffset startTimeExclusive;
         private DateTimeOffset endTimeInclusive;
 
-        internal AggregatorStore(AggregationType aggType, AggregationTemporality temporality)
+        internal AggregatorStore(
+            AggregationType aggType,
+            AggregationTemporality temporality,
+            string[] tagKeysInteresting = null)
         {
             this.metrics = new MetricPoint[MaxMetricPoints];
             this.aggType = aggType;
             this.temporality = temporality;
             this.startTimeExclusive = DateTimeOffset.UtcNow;
+            if (tagKeysInteresting != null)
+            {
+                var hs = new HashSet<string>(StringComparer.Ordinal);
+                foreach (var key in tagKeysInteresting)
+                {
+                    hs.Add(key);
+                }
+
+                this.tagKeysInteresting = hs;
+                this.tagsKeysInterestingCount = hs.Count;
+            }
+        }
+
+        internal void UpdateLong(long value, ReadOnlySpan<KeyValuePair<string, object>> tags)
+        {
+            try
+            {
+                var index = this.tagKeysInteresting != null ? this.FindMetricAggregatorIndexUserConfigured(tags) : this.FindMetricAggregatorIndexDefault(tags);
```

Comment:
We don't have check `this.tagKeysInteresting` each time because it can be determined ahead of time?

Reply:
The approach used here : https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/src/OpenTelemetry/Trace/TracerProviderSdk.cs#L35, was not doable here, as the argument to the function is a ref struct (ReadOnlySpan), so it cannot be declares as part of instance variable..

Need to some other tricks. Keeping this comment open now.

Reply:
@CodeBlanch shared an idea of using our own Delegate to overcome the limitation of ref structs..Exploring it now.

Reply:
Using delegate approach to avoid this perf hit : https://github.com/open-telemetry/opentelemetry-dotnet/pull/2442/files